pallets/elections/src/lib.rs

```diff
@@ -0,0 +1,171 @@
+    }
+
+    #[pallet::pallet]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::weight((T::BlockWeights::get().max_block, DispatchClass::Operational))]
+        pub fn change_members(origin: OriginFor<T>, members: Vec<T::AccountId>) -> DispatchResult {
+            ensure_root(origin)?;
+            Members::<T>::put(members.clone());
+            Self::deposit_event(Event::ChangeMembers(members));
+
+            Ok(())
+        }
+    }
+
+    #[pallet::type_value]
+    pub(super) fn DefaultForSessionsPerEra() -> u32 {
+        DEFAULT_SESSIONS_PER_ERA
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn sessions_per_era)]
+    pub(super) type SessionsPerEra<T: Config> =
+        StorageValue<_, SessionIndex, ValueQuery, DefaultForSessionsPerEra>;
+
+    #[pallet::type_value]
+    pub(super) fn DefaultForSessionPeriod() -> u32 {
+        DEFAULT_SESSION_PERIOD
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn session_period)]
+    pub(super) type SessionPeriod<T: Config> =
+        StorageValue<_, u32, ValueQuery, DefaultForSessionPeriod>;
+
+    #[pallet::type_value]
+    pub(super) fn DefaultForMillisecsPerBlock() -> u64 {
+        DEFAULT_MILLISECS_PER_BLOCK
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn millisecs_per_block)]
+    pub(super) type MillisecsPerBlock<T: Config> =
+        StorageValue<_, u64, ValueQuery, DefaultForMillisecsPerBlock>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        pub members: Vec<T::AccountId>,
+        pub millisecs_per_block: u64,
+        pub session_period: u32,
+        pub sessions_per_era: SessionIndex,
+    }
+
+    #[cfg(feature = "std")]
+    impl<T: Config> Default for GenesisConfig<T> {
+        fn default() -> Self {
+            Self {
+                members: Vec::new(),
+                millisecs_per_block: DEFAULT_MILLISECS_PER_BLOCK,
+                session_period: DEFAULT_SESSION_PERIOD,
+                sessions_per_era: DEFAULT_SESSIONS_PER_ERA,
+            }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+        fn build(&self) {
+            <Members<T>>::put(&self.members);
+            <MillisecsPerBlock<T>>::put(&self.millisecs_per_block);
+            <SessionPeriod<T>>::put(&self.session_period);
+            <SessionsPerEra<T>>::put(&self.sessions_per_era);
+        }
+    }
+
+    impl<T: Config> Pallet<T> {}
+
+    #[derive(Debug)]
+    pub enum Error {
+        DataProvider(&'static str),
+    }
+
+    impl<T: Config> ElectionProvider<T::AccountId, BlockNumberFor<T>> for Pallet<T> {
+        type Error = Error;
+        type DataProvider = T::DataProvider;
+
+        // We use authority list so the supports don't influance the result
+        // but we still have to calculate them for the sake of eras payouts
```

Comment:
This comment is rather confusing. It should say something along the lines of:
"As of now the election results do not influence the list of validators, which is based on a sudo-managed authority list. However, we still need to calculate the `Supports` for the sake of era payouts."

Reply:
But your version is not quite right. For every new era pallet staking uses this `elect` method to get new validators, so the election results do influence the list of validators. What is missing in this scheme is some `vote` extrinsic. What we have is just `change_members` sudo call.
Which part is confusing? I can try to improve it.

Reply:
OK, so apparently I did not understand how it works. But what confuses me in this comment is saying that `supports` don't influence (also note typo) the `result`. Which result? If this is the result of the function, then well, the result is `supports` :P

Reply:
I meant the results of the elections :)

Reply:
I think the nomenclature here is that `Supports` is the result of elections.

Reply:
You're right. How about:
// The elections are PoA so only the nodes listed in the Members will be elected as validators.
// We calculate the supporst for them for the sake of eras payouts.

Reply:
Great. A small typo only: `supporst`